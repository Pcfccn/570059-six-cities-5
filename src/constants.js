const Path = {
  MAIN: `/`,
  SIGN_IN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer/:id`,
  getUrlById: (id) => (`/offer/${id}`)
};

const ratingStars = [5, 4, 3, 2, 1];
const ratingInputTitle = [`perfect`, `good`, `not bad`, `badly`, `terribly`];

const widthStartsCoefficient = 20;

const CityCoordinates = {
  Amsterdam: [52.38333, 4.9],
  Paris: [48.864716, 2.349014],
};

const OfferCardClassName = {
  nearPlace: `near-places__card`,
  citiesPlace: `cities__place-card`,
};

const MapClassName = {
  property: `property__map`,
  cities: `cities__map`,
};

const cities = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export {Path, widthStartsCoefficient, ratingStars, ratingInputTitle, CityCoordinates, OfferCardClassName, MapClassName, cities};
