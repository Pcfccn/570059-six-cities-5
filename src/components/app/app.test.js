import React from "react";
import {Router} from "react-router-dom";
import App from "./app";
import browserHistory from "../../brouser-history";
import {Provider} from "react-redux";
import toJson from "enzyme-to-json";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureMockStore from 'redux-mock-store';
import {AuthorizationStatus} from "../../constants";

configure({adapter: new Adapter()});

let mockState;
beforeEach(() => {
  mockState = {
    USER: {
      authorizationStatus: AuthorizationStatus.AUTH,
      email: `sdf@mail.ri`
    },
    STATE: {
      city: `Paris`
    },
    DATA: {
      offers: [],
    }
  };
});
it(`Should App render correctly`, () => {
  const mockStore = configureMockStore();
  const store = mockStore(mockState);
  const wrapper = mount(
      <Provider store={store}>
        <Router history={browserHistory}>
          <App />
        </Router>
      </Provider>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
