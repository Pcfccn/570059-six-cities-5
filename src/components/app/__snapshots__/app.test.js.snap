// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should App render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Switch>
          <Route
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
            render={[Function]}
          >
            <Connect(Main)>
              <Main
                city="Paris"
                dispatch={[Function]}
                offers={Array []}
              >
                <div
                  className="page page--gray page--main"
                >
                  <Connect(Header)>
                    <Header
                      authStatus="AUTH"
                      dispatch={[Function]}
                      email="sdf@mail.ri"
                    >
                      <header
                        className="header"
                      >
                        <div
                          className="container"
                        >
                          <div
                            className="header__wrapper"
                          >
                            <div
                              className="header__left"
                            >
                              <Link
                                className="header__logo-link"
                                to="/"
                              >
                                <LinkAnchor
                                  className="header__logo-link"
                                  href="/"
                                  navigate={[Function]}
                                >
                                  <a
                                    className="header__logo-link"
                                    href="/"
                                    onClick={[Function]}
                                  >
                                    <img
                                      alt="6 cities logo"
                                      className="header__logo"
                                      height="41"
                                      src="/img/logo.svg"
                                      width="81"
                                    />
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </div>
                            <nav
                              className="header__nav"
                            >
                              <ul
                                className="header__nav-list"
                              >
                                <li
                                  className="header__nav-item user"
                                >
                                  <Link
                                    className="header__nav-link header__nav-link--profile"
                                    to="/favorites"
                                  >
                                    <LinkAnchor
                                      className="header__nav-link header__nav-link--profile"
                                      href="/favorites"
                                      navigate={[Function]}
                                    >
                                      <a
                                        className="header__nav-link header__nav-link--profile"
                                        href="/favorites"
                                        onClick={[Function]}
                                      >
                                        <div
                                          className="header__avatar-wrapper user__avatar-wrapper"
                                        />
                                        <span
                                          className="header__user-name user__name"
                                        >
                                          sdf@mail.ri
                                        </span>
                                      </a>
                                    </LinkAnchor>
                                  </Link>
                                </li>
                              </ul>
                            </nav>
                          </div>
                        </div>
                      </header>
                    </Header>
                  </Connect(Header)>
                  <main
                    className="page__main page__main--index"
                  >
                    <h1
                      className="visually-hidden"
                    >
                      Cities
                    </h1>
                    <Connect(CitiesList)>
                      <CitiesList
                        changeCity={[Function]}
                        city="Paris"
                      >
                        <div
                          className="tabs"
                        >
                          <section
                            className="locations container"
                          >
                            <ul
                              className="locations__list tabs__list"
                            >
                              <City
                                active={true}
                                changeCity={[Function]}
                                key="Paris"
                                thisCity="Paris"
                              >
                                <li
                                  className="locations__item"
                                >
                                  <a
                                    className="locations__item-link tabs__item  tabs__item--active"
                                    onClick={[Function]}
                                  >
                                    <span>
                                      Paris
                                    </span>
                                  </a>
                                </li>
                              </City>
                              <City
                                active={false}
                                changeCity={[Function]}
                                key="Cologne"
                                thisCity="Cologne"
                              >
                                <li
                                  className="locations__item"
                                >
                                  <a
                                    className="locations__item-link tabs__item "
                                    onClick={[Function]}
                                  >
                                    <span>
                                      Cologne
                                    </span>
                                  </a>
                                </li>
                              </City>
                              <City
                                active={false}
                                changeCity={[Function]}
                                key="Brussels"
                                thisCity="Brussels"
                              >
                                <li
                                  className="locations__item"
                                >
                                  <a
                                    className="locations__item-link tabs__item "
                                    onClick={[Function]}
                                  >
                                    <span>
                                      Brussels
                                    </span>
                                  </a>
                                </li>
                              </City>
                              <City
                                active={false}
                                changeCity={[Function]}
                                key="Amsterdam"
                                thisCity="Amsterdam"
                              >
                                <li
                                  className="locations__item"
                                >
                                  <a
                                    className="locations__item-link tabs__item "
                                    onClick={[Function]}
                                  >
                                    <span>
                                      Amsterdam
                                    </span>
                                  </a>
                                </li>
                              </City>
                              <City
                                active={false}
                                changeCity={[Function]}
                                key="Hamburg"
                                thisCity="Hamburg"
                              >
                                <li
                                  className="locations__item"
                                >
                                  <a
                                    className="locations__item-link tabs__item "
                                    onClick={[Function]}
                                  >
                                    <span>
                                      Hamburg
                                    </span>
                                  </a>
                                </li>
                              </City>
                              <City
                                active={false}
                                changeCity={[Function]}
                                key="Dusseldorf"
                                thisCity="Dusseldorf"
                              >
                                <li
                                  className="locations__item"
                                >
                                  <a
                                    className="locations__item-link tabs__item "
                                    onClick={[Function]}
                                  >
                                    <span>
                                      Dusseldorf
                                    </span>
                                  </a>
                                </li>
                              </City>
                            </ul>
                          </section>
                        </div>
                      </CitiesList>
                    </Connect(CitiesList)>
                    <MainNoPlaces
                      city="Paris"
                    >
                      <div
                        className="cities page__main--index-empty"
                      >
                        <div
                          className="cities__places-container cities__places-container--empty container"
                        >
                          <section
                            className="cities__no-places"
                          >
                            <div
                              className="cities__status-wrapper tabs__content"
                            >
                              <b
                                className="cities__status"
                              >
                                No places to stay available
                              </b>
                              <p
                                className="cities__status-description"
                              >
                                We could not find any property available at the moment in 
                                Paris
                              </p>
                            </div>
                          </section>
                          <div
                            className="cities__right-section"
                          />
                        </div>
                      </div>
                    </MainNoPlaces>
                  </main>
                </div>
              </Main>
            </Connect(Main)>
          </Route>
        </Switch>
      </Router>
    </App>
  </Router>
</Provider>
`;
